partiql-ast/src/main/resources/partiql_ast.ion

```diff
@@ -196,10 +196,29 @@ type::[
   timestamp_with_tz::{ precision: optional::int },             // TIMESTAMP [(<int>)] WITH TIMEZONE
   interval::{ precision: optional::int },                      // INTERVAL
   bag::{},                                                     // BAG
+  // Keep the list type for backward compatible reason
+  // user should migrate to the array type
   list::{},                                                    // LIST
+  array::{ type: optional::type },                             // ARRAY
   sexp::{},                                                    // SEXP
+  // Keep the tuple type for backward compatible reason
+  // migration: in v0.15.0: struct modeling should NOT be Introduced,
+  //  users of struct type should be migrate to tuple type.
+  //  in v1.0.0: introduce the struct modeling change, all users from tuple type should migrate to struct type.
   tuple::{},                                                   // TUPLE
-  struct::{},                                                  // STRUCT
+  struct::{
+    fields: list::[field],
```

Comment:
So is the list of fields always required for structs? It appears somewhat different than the other types with parameters, which are all optional parameters.

Reply:
Wondering if there is a distinction between a struct with unspecified fields and a struct with no specified fields (i.e. empty list)?

Reply:
`Struct<>` and `LIST<>` is not permitted by the ANTLR Grammar to avoid this confusion.

Reply:
From offline discussion, the semantics for `STRUCT<>` and `LIST<>` is currently undefined. Can decide in the future whether this
- is allowed
- if allowed, does it have different semantics from `STRUCT` and `LIST` (without any fields or element specified within the angle brackets)